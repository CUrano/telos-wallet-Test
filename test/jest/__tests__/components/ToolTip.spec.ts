import { shallowMount } from '@vue/test-utils';
import ToolTip from 'components/ToolTip.vue';
import { stubWithSlot } from 'test/jest/testing-helpers';

describe('ToolTip.vue', () => {
    it('should render correctly', async () => {
        const wrapper = shallowMount(ToolTip, {
            global: {
                stubs: {
                    'q-tooltip': stubWithSlot('q-tooltip'),
                },
            },
            props: {
                text: 'Test',
            },
        });

        // no warning text
        expect(wrapper.element).toMatchSnapshot();

        await wrapper.setProps({
            warnings: ['test', 'test2'],
        });

        // with warning text
        expect(wrapper.element).toMatchSnapshot();
    });
});
