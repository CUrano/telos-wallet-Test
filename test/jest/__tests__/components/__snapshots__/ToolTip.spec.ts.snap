// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolTip.vue should render correctly 1`] = `
<div
  class="c-tooltip"
>
  <inline-svg-stub
    aria-hidden="true"
    class="c-tooltip__icon"
    keepduringloading="true"
    src=""
    transformsource="[Function]"
  />
  <q-tooltip-stub
    hide-delay="5000"
    transition-hide="scale"
    transition-show="scale"
  >
    
    <div
      class="c-tooltip__text-container"
    >
      
      
      <span
        class="c-tooltip__text"
      >
        Test
      </span>
    </div>
    
  </q-tooltip-stub>
</div>
`;

exports[`ToolTip.vue should render correctly 2`] = `
<div
  class="c-tooltip"
>
  <inline-svg-stub
    aria-hidden="true"
    class="c-tooltip__icon"
    keepduringloading="true"
    src=""
    transformsource="[Function]"
  />
  <q-tooltip-stub
    hide-delay="5000"
    transition-hide="scale"
    transition-show="scale"
  >
    
    <div
      class="c-tooltip__text-container"
    >
      
      <div
        class="c-tooltip__warning-container"
      >
        <inline-svg-stub
          aria-hidden="true"
          class="c-tooltip__icon c-tooltip__icon--warning"
          keepduringloading="true"
          src=""
          transformsource="[Function]"
        />
        <span
          class="c-tooltip__text"
        >
          test
        </span>
      </div>
      <div
        class="c-tooltip__warning-container"
      >
        <inline-svg-stub
          aria-hidden="true"
          class="c-tooltip__icon c-tooltip__icon--warning"
          keepduringloading="true"
          src=""
          transformsource="[Function]"
        />
        <span
          class="c-tooltip__text"
        >
          test2
        </span>
      </div>
      
      <span
        class="c-tooltip__text"
      >
        Test
      </span>
    </div>
    
  </q-tooltip-stub>
</div>
`;
